k8_cortx_cloud/prereq-deploy-cortx-cloud.sh

```diff
@@ -150,6 +274,118 @@ function prepCortxDeployment()
     done
 }
 
+function join_array()
+{
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+
+## Function 'symlinkBlockDevices' - enabled via the '-b' flag
+## ----------------------------------------------------------
+##      Create symlinks for persistent block device access, based
+##      upon device paths defined in 'solution.yaml' and the 
+##      symlink path separator defined with the '-c' option.
+##      As an example, '/dev/sdc' will have a new symlink available via
+##      '/dev/cortx/sdc' for persistent access across reboots.
+##      This option will also create an updated '{solution}-symlink.yaml'
+##      file that should be used for subsequent deployment with 
+##      'deploy-cortx-cloud.sh'.
+## 
+##      Background
+##      - This implementation is meant to be invoked from a control-plane 
+##        node with local kubectl accessand requires parsing node information
+##        from solution.yaml
+##      - An alternative implementation would be to be run on the individual
+##        node itself, using NODE_NAME=$(hostname) with no need for looping
+##        through nodes defined in solution.yaml, but requiring local kubectl access.
+function symlinkBlockDevices()
+{
+    printf "####################################################\n"
+    printf "# Create Block Device Symlinks                      \n"
+    printf "####################################################\n"
+
+    # Local variables
+    local filter
+    local node_list=()
+    local device_paths=()
+    local job_template="$(pwd)/cortx-cloud-3rd-party-pkg/templates/job-symlink-block-devices.yaml.template"
+    local template_vars='${NODE_NAME}:${NODE_SHORT_NAME}:${DEVICE_PATHS}:${SYMLINK_PATH_SEPARATOR}:${CORTX_IMAGE}'
+    local job_file="jobs-symlink-block-devices-$(hostname | cut -f1 -d'.').yaml"
+
+    # Template replacement variable
+    export SYMLINK_PATH_SEPARATOR=${symlink_block_devices_separator}
+
+    # Retrieve CORTX container image specified in solution.yaml
+    filter="solution.images.cortxcontrol"
+    cortx_image_yaml=$(parseSolution ${solution_yaml} ${filter})
+    export CORTX_IMAGE=$(echo ${cortx_image_yaml} | cut -f2 -d'>')
+    
+    # Create comma-separated string from the device paths in solution.yaml
+    filter="solution.storage.cvg*.devices*.device"
+    device_output=$(parseSolution ${solution_yaml} ${filter})
+    IFS=';' read -r -a device_array <<< "${device_output}"
+    for device_element in "${!device_array[@]}"
+    do
+        device_paths+=($(echo ${device_array[${device_element}]} | cut -f2 -d'>')) 
```

Comment:
```suggestion
    for device_path in "${device_array[@]}"
    do
        device_paths+=(${device_path#*>})
```
Can iterate over array elements, index is not used for anything. Path can be split with pure bash.